cpu/efm32_common/periph/spi.c

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2014-2016 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_efm32_common
+ *
+ * @{
+ *
+ * @file
+ * @brief       Low-level SPI driver implementation
+ *
+ * @author      Ryan Kurte <ryankurte@gmail.com>
+ * @author      Bas Stottelaar <basstottelaar@gmail.com>
+ * @author      Christian Amsüss <c@amsuess.com>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "sched.h"
+#include "thread.h"
+#include "mutex.h"
+
+#include "periph_conf.h"
+#include "periph/gpio.h"
+#include "periph/spi.h"
+
+#include "em_device.h"
+#include "em_cmu.h"
+#include "em_usart.h"
+
+/* guard file in case no SPI device is defined */
+#if SPI_NUMOF
+
+static mutex_t spi_lock[SPI_NUMOF] = {
+#if SPI_0_EN
+    [0] = MUTEX_INIT,
+#endif
+#if SPI_1_EN
+    [1] = MUTEX_INIT,
+#endif
```

Comment:
same here, simply call mutex_init in the spi_init function -> much nicer to read and scales better :-)

Reply:
Fixed.